src/Compilers/CSharp/Portable/Syntax/LookupPosition.cs

```diff
@@ -289,7 +290,8 @@ private static SyntaxToken GetFirstIncludedToken(StatementSyntax statement)
                 case SyntaxKind.FixedStatement:
                     return ((FixedStatementSyntax)statement).FixedKeyword;
                 case SyntaxKind.ForEachStatement:
-                    return ((ForEachStatementSyntax)statement).OpenParenToken.GetNextToken();
+                case SyntaxKind.ForEachComponentStatement:
```

Comment:
>ForEachComponentStatement [](start = 32, length = 25)

Did you search on `SyntaxKind.ForEachStatement` to confirm no places where missed that should also handle `ForEachComponentStatement`?